src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c

```diff
@@ -110,6 +117,90 @@ static void zchan_cmd_identify(struct zocl_rpu_channel *chan, struct xgq_cmd_sq_
 	r->minor = ZCHAN_CMD_HANDLER_VER_MINOR;
 }
 
+static void zchan_cmd_log_page(struct zocl_rpu_channel *chan, struct xgq_cmd_sq_hdr *cmd,
+	struct xgq_com_queue_entry *resp)
+{
+	struct xgq_cmd_sq *sq = (struct xgq_cmd_sq *)cmd;
+	struct xgq_cmd_cq *cq = (struct xgq_cmd_cq *)resp;
+	u32 address_offset = sq->log_payload.address;
+	u32 size = sq->log_payload.size;
+	u32 offset = sq->log_payload.offset;
+	int pid = sq->log_payload.pid;
+	struct zocl_rpu_data_entry *entry = NULL;
+	void __iomem *src = chan->mem_base + address_offset;
+
+	zchan_info(chan, "addr_off 0x%x, size %d, offset %d, pid 0x%x",
+		address_offset, size, offset, pid);
+
+	if (pid != XGQ_CMD_LOG_XCLBIN) {
+		zchan_err(chan, "unsupported type 0x%x", pid);
+		goto fail;
+	}
+
+	/*
+	 * if offset is not 0, this is not last pkt.
+	 *
+	 * every pkt will be added to linkedlist.
+	 * if this is last pkt, dump linked list to whole xclbin.
+	 */
+	entry = vmalloc(sizeof(struct zocl_rpu_data_entry));
+	entry->data_size = size;
+	entry->data_entry = vmalloc(size);
+	memcpy_fromio(entry->data_entry, src, size);
+
+	list_add_tail(&entry->entry_list, &chan->data_list);
+
+	/* offset 0 indicates this is the last pkt */
+	if (offset == 0) {
```

Comment:
what is the offset value for the very first pkt? it is not 0?

Reply:
you mean whatif the offset is not 0 for the first pkg?

it will add into the list and return.
The VMR code will send next request, till offset is 0.
https://github.com/Xilinx/VMR/pull/52

This is the basic protocol of sending multiple trunks.